app/models/smart_proxy.rb

```diff
@@ -100,6 +100,16 @@ def setting(feature, setting)
     smart_proxy_features.find_by(:feature_id => Feature.find_by(:name => feature)).try(:settings).try(:[], setting)
   end
 
+  # Smart proxy exposed settings may contain sensitive data
+  ALLOWED_SETTINGS = [
+    "http_port",
+    "https_port",
+  ]
+  def safe_setting(feature, setting)
+    raise ::Foreman::Exception.new(N_("Setting is not whitelisted: %s"), ALLOWED_SETTINGS) unless ALLOWED_SETTINGS.include?(setting)
```

Comment:
This shows what setting are allowed, but not which setting was actually requested
```ruby
::Foreman::Exception.new(N_("Setting is not whitelisted: %s"), ALLOWED_SETTINGS).message
=> "ERF42-4295 [Foreman::Exception]: Setting is not whitelisted: [\"http_port\", \"https_port\"]"
```

Reply:
Fixing and adding tests for this.